src/txn/txn_rollback_to_stable.c

```diff
@@ -145,6 +148,45 @@ __rollback_row_add_update(WT_SESSION_IMPL *session, WT_PAGE *page, WT_ROW *rip,
     return (ret);
 }
 
+/*
+ * __rollback_check_if_txnid_non_committed --
+ *     Check if the transaction id is non committed.
```

Comment:
What has changed to require us to use this helper everywhere?

Reply:
With the current change, the rollback to stable will check for the transaction id in addition to the stable timestamp check to rollback any update. This helper function is added to check the transaction id with the recovered snapshot details and find if the transaction is committed or not.